### PR TITLE
Kconfig: drop `COMPAT_INCLUDES`

### PR DESCRIPTION
**Pull request by ycsin**
`COMPAT_INCLUDES` was introduced a few years ago (#21326 v2.2.0) to suppress compiler warnings when using old header location. These headers shim have since been removed from tree and no source in tree refers to `CONFIG_COMPAT_INCLUDES` anymore.

Dropping it as it is not used.

### CODE REVIEW

**Review by cfriedt — DISMISSED** [review]

**Review by carlescufi — APPROVED** [review]

**Review by carlescufi — CHANGES_REQUESTED** [review]
The commit message title is incorrect

**Review by nashif — DISMISSED** [review]

**Comment by ycsin** [comment]
fixed commit title

**Comment by nashif** [comment]
You are leaving an empty menu section after this kconfig is removed, it will need to be removed....
Actually, isn't this something that you should use for something like https://github.com/zephyrproject-rtos/zephyr/pull/63973? Why introduce new Kconfigs for compatibility if this can be used?

**Comment by ycsin** [comment]
> Why introduce new Kconfigs for compatibility if this can be used?

I didn't realize this Kconfig when I put up #63973 initially, it was basically referencing #44320, with some changes to the descriptions, `COMPAT_INCLUDES` can probably be repurposed for #63973

**Review by carlescufi — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Comment by ycsin** [comment]
This can probably wait until #63973 is merged

**Review by cfriedt — APPROVED** [review]

**Comment by ycsin** [comment]
superseded by #64957

**Pull request closed without merging**
